src/hsmm_mvpy/models.py

```diff
@@ -160,6 +177,15 @@ def __init__(self, data, epoch_data=None, sfreq=None, cpus=1, event_width=50, sh
             self.convolution = np.convolve
         self.trial_coords = data.unstack().sel(component=0,samples=0).rename({'epochs':'trials'}).\
             stack(trial_x_participant=['participant','trials']).dropna(dim="trial_x_participant",how="all").coords
+        if epoch_data is not None:
+            if len(epoch_data.dims) == 4:
+                self.stacked_epoch_data = epoch_data.rename({'epochs':'trials'}).stack(trial_x_participant=('participant','trials')).data.fillna(0).drop_duplicates('trial_x_participant')
```

Comment:
I first did it on components, but especially with few components it is much less stable, and also less clear what thresholds are needed. I really wanted to avoid using the original channel data, but it made it all much more stable.

Reply:
Perhaps we can include 'epoch_data' by default in HMP, and recommend deleting it from the Python environment in general. Not optimal, but given that we need epoch_data anyway for plotting (which we can then just integrate) it's typically in RAM.